source/Octopus.Tentacle.Tests/Contracts/Legacy/CanSerializeAndDeserializeJsonWithLegacyNameSpace.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.IO;
+using FluentAssertions;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.Legacy;
+using Octopus.Tentacle.Tests.Util;
+
+namespace Octopus.Tentacle.Tests.Contracts.Legacy
+{
+    public class CanSerializeAndDeserializeJsonWithLegacyNameSpace
```

Comment:
This adds a test which shows that the json serialisation actually works as expected.

Reply:
Can we make sure all the legacy types serialise and de-serialise correctly

Reply:
Added tests in both directions for all the concrete classes.

I don't think the interfaces are actually serialised since Halibut seems to send just the name (short name of the class without Namespace) and the method name.